vulyk/blueprints/gamification/core/queries.py

```diff
@@ -59,14 +59,15 @@ def __init__(self, rule: Rule) -> None:
         """
         super().__init__()
 
+        self._filter_first['answer'] = {'$exists': True}
         if isinstance(rule, ProjectRule):
             self._filter_first['taskType'] = rule.task_type_name
 
         # we filter out tasks older than given date in these cases:
         # - n tasks in m days
-        is_tasks_in_days = rule.days_number > 0 and rule.tasks_number > 0
+        is_tasks_in_days = (rule.days_number or 0) > 0 and (rule.tasks_number or 0) > 0
```

Comment:
What are those fall-backs for? Wouldn't it be more sound to prevent creating malformed rules in the constructor?
Also I see a tiny bug [here](https://github.com/mrgambal/vulyk/blob/master/vulyk/blueprints/gamification/core/parsing.py#L56). It seems to me that the implicit integer coercion has been omitted.